### PR TITLE
For #14565: Add telemetry for top sites

### PR DESCRIPTION
**Pull request by sblatz**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by sblatz** [comment]
# Request for data collection review form

**All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.**

1) **What questions will you answer with this data?**
- How do users interact with different types of top sites (frecency, pinned, default)?
- Do users swipe on the carousel (are they aware of paginated top sites)

2) **Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements?**

To measure how useful the different types of top sites are to users. It will help us prioritize top sites features, and help us understand if the carousel for top sites is understood by users or if we should reconsider that UX.

3) **What alternative methods did you consider to answer these questions? Why were they not sufficient?**

* We have events for other types of top sites actions, but none that distinguish between the *types* of top sites. Nor do we have any events for the carousel itself, which will be really helpful for us.

4) **Can current instrumentation answer these questions?**

* No, see above.

5) **List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox [data c](https://wiki.mozilla.org/Firefox/Data_Collection)[ollection ](https://wiki.mozilla.org/Firefox/Data_Collection)[categories](https://wiki.mozilla.org/Firefox/Data_Collection) on the found on the Mozilla wiki.**

* All data is Category 2.

6) **How long will this data be collected?**

This data will be present collected until it expires as outlined in the `metrics.md` file.

7) **What populations will you measure?**

* All release, beta, and nightly users with telemetry enabled.

8) **Please provide a general description of how you will analyze this data.**

* Leanplum / Amplitude

9) **Where do you intend to share the results of your analysis?**

* Only on Leanplum, Amplitude and with mobile teams.

**Review by gabrielluong — APPROVED** [review]

File: app/metrics.yaml [hunk 1]
```diff
@@ -3195,6 +3195,66 @@ top_sites:
...
       - fenix-core@mozilla.com
     expires: "2020-11-15"
+  open_frecency:
+    type: event
+    description: |
+      A user opened a frecency top site
+    bugs:
+      - https://github.com/mozilla-mobile/fenix/issues/14565
+    data_reviews:
+      - https://github.com/mozilla-mobile/fenix/pull/TODO
```
Comment (liuche) [inline app/metrics.yaml:3205]: TODO

File: app/metrics.yaml [hunk 2]
```diff
@@ -3195,6 +3195,66 @@ top_sites:
...
+    data_sensitivity:
+      - interaction
+    notification_emails:
+      - fenix-core@mozilla.com
+    expires: "2020-12-15"
```
Comment (liuche) [inline app/metrics.yaml:3210]: Could you move these to 3/15 since that will be 6mo from now?

**Review by liuche — APPROVED** [review]
# Data Review Form (to be filled by Data Stewards)

1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way? 

Yes, metrics.yaml

2) Is there a control mechanism that allows the user to turn the data collection on and off? 

Yes, fenix telemetry settings

3) If the request is for permanent data collection, is there someone who will monitor the data over time?

No has expiry

4) Using the **[category system of data types](https://wiki.mozilla.org/Firefox/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?

Type 2, interaction with top sites and the carousel

5) Is the data collection request for default-on or default-off?

default on

6) Does the instrumentation include the addition of **any *new* identifiers** (whether anonymous or otherwise; e.g., username, random IDs, etc.  See the appendix for more details)?

No, interaction with UI elements

7) Is the data collection covered by the existing Firefox privacy notice?

yes

8) Does there need to be a check-in in the future to determine whether to renew the data? (Yes/No) (If yes, set a todo reminder or file a bug if appropriate)**

No, has expiry

9) Does the data collection use a third-party collection tool? **If yes, escalate to legal.**

No

File: app/metrics.yaml (hunk 2 above)
Comment (sblatz) [reply app/metrics.yaml]: Done!

**Pull request merged**